dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/BaseService.java

```diff
@@ -76,6 +76,14 @@ public interface BaseService {
      */
     void putMsg(Result<Object> result, Status status, Object... statusParams);
 
+    /**
+     * copy result msg from source to target
+     *
+     * @param        source source result
+     * @param        target target result
+     */
+    void copyMsg(Result source, Result target);
+
```

Comment:
What is this used for?

Reply:
I  use this for copy msg Result from one service-method to another Result,because i think the result should declare  generic type ，but we usually have different generic type ，and the msg is must delived by result ，the code below show this，
```java
        Result<Resource> result = new  Result<>();
       ........
       .......
       Result<Object> verifyPidResult = verifyPid(loginUser, pid);
        if (!verifyPidResult.getCode().equals(Status.SUCCESS.getCode())) {
            copyMsg(verifyPidResult, result);
            return result;
        }
```
of course i can set /get code and msg in every time

Reply:
> I think it's unnessnary since we are doing V2 refactoring.

+1, I think we don't need to copy result, because the old way is bad case, which should be optimized in the future.

Reply:
yes ,i agree with you .but i just don't kown how to do with it  for now,using get/set ?

Reply:
> yes ,i agree with you .but i just don't kown how to do with it for now,using get/set ?

You can take a look at `org.apache.dolphinscheduler.api.controller.AccessTokenV2Controller`.

Reply:
i see, but  it's different , AccessTokenServiceImpl don't use The generic-type, and it is too simple, just use putMsg。

Reply:
@SbloodyS @caishunfeng  so what i should do is  delete copy method and use get/set ？ right ？

Reply:
> @SbloodyS @caishunfeng so what i should do is delete copy method and use get/set ？ right ？

Yes.

Reply:
> 

OK ，I fix it and repush， approve please